tests/conftest.py

```diff
@@ -197,3 +198,23 @@ def selector_loop() -> None:
     with loop_context(policy.new_event_loop) as _loop:
         asyncio.set_event_loop(_loop)
         yield _loop
+
+
+@pytest.fixture
+def netrc_contents(
+    tmp_path: pathlib.Path,
+    monkeypatch: pytest.MonkeyPatch,
+    request: pytest.FixtureRequest,
+):
+    """
+    Prepare a ``.netrc`` file with given contents.
+
+    monkeypatches the NETRC environment variable to point to created
+    file.
+    """
+    netrc_contents = request.param
+
+    netrc_file_path = tmp_path / ".netrc"
+    netrc_file_path.write_text(netrc_contents)
+
+    monkeypatch.setenv("NETRC", str(netrc_file_path))
```

Comment:
It could be useful to return the path too, so it could be used for populating the contents:
```suggestion
    monkeypatch.setenv("NETRC", str(netrc_file_path))

    return netrc_file_path
```